mesonbuild/modules/unstable_xorg.py

```diff
@@ -0,0 +1,140 @@
+# SPDX-License-Identifier: Apache-2.0
+# Copyright © 2021 Intel Corporation
+
+from mesonbuild.interpreterbase.decorators import KwargInfo, typed_kwargs
+import os
+import typing as T
+
+from . import NewExtensionModule, ModuleReturnValue
+from .. import mesonlib
+from ..build import InvalidArguments, CustomTarget
+from ..interpreterbase import FeatureNew
+from ..interpreterbase import typed_pos_args
+
+if T.TYPE_CHECKING:
+    from . import ModuleState
+    from ..interpreter import Interpreter
+
+    from typing_extensions import TypedDict
+
+    class FormatManKW(TypedDict):
+
+        apploaddir: T.Optional[str]
+
+
+__all__ = ['initialize']
+
+# This has been vastily simplified to only be accurate for xorg's purposes. If
+# this were to be used generally there would be bugs. In particualar the "sysv"
+# section is more accurately "old solaris/sunos"
+_MAN_SECTIONS = {
+    'traditional': {
+        'app': '1',
+        'driver': '4',
+        'admin': '8',
+        'lib': '3',
+        'misc': '7',
+        'file': '5',
+    },
+    'sysv': {
+        'app': '1',
+        'driver': '7',
+        'admin': '1m',
+        'lib': '3',
+        'misc': '5',
+        'file': '4',
+    }
+}
+
+
+class XorgModule(NewExtensionModule):
+
+    """Module for upstream xorg development.
+
+    This module contains helper and abstractions fo upstream xorg projects to
+    use. These are *not* meant for projects using X11, though they might be
+    useful in that case.
+    """
+
+    def __init__(self) -> None:
+        super().__init__()
+        self.methods.update({
+            'format_man': self.format_man_method,
+        })
+
+    def _uses_sysv_man_sections(self) -> bool:
+        """Returns true if using sysv man sections."""
+        # XXX: This is only correct for build == host
+        # Copied from xorg-macros
+        return os.path.exists('/usr/share/man/man7/attributes.7')
```

Comment:
Is this really the only way to detect the condition? Seems risky. A project option would be better.

Reply:
I don't know if it's even really useful to do this, @alanc would know I think. you really don't want a project optinon because in 99% of cases you don't want the old solaris sections.

Reply:
In theory what @alanc says is true, but in practice most OSS have not bothered in a while so we have it mixed now. FreeIPMI misc man page for example is in section 7 same as on Linux. So just update it and we will fix our own pages eventually.

For Uname checks that need to identify illumos use `uname -o`